contracts/core/PositionVault.sol

```diff
@@ -248,14 +248,15 @@ contract PositionVault is Constants, ReentrancyGuard, IPositionVault {
             _posId,
             false
         );
+        uint256 bounty = (marginFees * settingsManager.bountyPercent()) / BASIS_POINTS_DIVISOR;
         require(liquidationState != LIQUIDATE_NONE_EXCEED, "not exceed or allowed");
         if (liquidationState == LIQUIDATE_THRESHOLD_EXCEED) {
             // max leverage exceeded but there is collateral remaining after deducting losses so decreasePosition instead
+            vault.transferBounty(msg.sender, bounty);
```

Comment:
well, need to re-write bounty logic for this scenario, please rebase on the latest branch

Reply:
I will merge this as it fixes the bugs / issues, we can then rewrite based on the bounty, resolver and so on as discussed in a new PR @mookim-eth